commit 6259c0b909a8c00fabe3c7e6bd81150ee08cbf9f
Merge: 31afda3 630ff31
Author: Mike Pall <mike>
Date:   Wed May 17 17:38:53 2017 +0200

    Merge branch 'master' into v2.1

commit 630ff3196a06353c6a7ccd1e9ac3958f4a8ca13c
Author: Mike Pall <mike>
Date:   Wed May 17 17:37:35 2017 +0200

    Add missing LJ_MAX_JSLOTS check.
    
    Thanks to Yichun Zhang.

From 630ff3196a06353c6a7ccd1e9ac3958f4a8ca13c Mon Sep 17 00:00:00 2001
From: Mike Pall <mike>
Date: Wed, 17 May 2017 17:37:35 +0200
Subject: [PATCH 02/72] Add missing LJ_MAX_JSLOTS check.

Thanks to Yichun Zhang.
---
 src/lj_record.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lj_record.c b/src/lj_record.c
index cecacd2..bc4e8a6 100644
--- a/src/lj_record.c
+++ b/src/lj_record.c
@@ -633,6 +633,8 @@ void lj_record_call(jit_State *J, BCReg func, ptrdiff_t nargs)
   J->framedepth++;
   J->base += func+1+LJ_FR2;
   J->baseslot += func+1+LJ_FR2;
+  if (J->baseslot + J->maxslot >= LJ_MAX_JSLOTS)
+    lj_trace_err(J, LJ_TRERR_STACKOV);
 }
 
 /* Record tail call. */
-- 
2.20.1
